packages/drivers/routerlicious-driver/src/wholeSummaryDocumentStorageService.ts

```diff
@@ -86,14 +85,60 @@ export class WholeSummaryDocumentStorageService implements IDocumentStorageServi
 		// If this is the first versions call for the document, we know we will want the latest summary.
 		// Fetch latest summary, cache it, and return its id.
 		if (this.firstVersionsCall && count === 1) {
+			const normalizedSnapshotContents = await PerformanceEvent.timedExecAsync(
+				this.logger,
+				{
+					eventName: "ObtainSnapshot",
+					versionId: versionId ?? undefined,
+					count,
+					enableDiscovery: this.driverPolicies?.enableDiscovery,
+				},
+				async (event) => {
+					let method: string;
+					const cachedSnapshotP = this.snapshotTreeCache.get(
```

Comment:
While this is likely fastest way to get caching going, I worry about not doing it in shared way (i.e. outside the driver).
It would force us to think about many things that we might miss otherwise.
For example, I think we need to add various policies here - not every user would be happy with cache pros/cons (though maybe this can be alleviated by users controlling presence of cache like in ODSP case - I do not think AFR stack gives that flexibility)

Reply:
That said, if I get it right, this cache is in-memory cache, and only helps summarizer? I.e. it does not survive app reload, right?

Reply:
Yes, on reload we will get different cache. However, in our stress tests we do load several clients which would also be benefitted. As far as adding policies, that can be done in separate PR later on as the logic is not changed here as what was before when cache is always present.